commands/publish_dafsa.py

```diff
@@ -0,0 +1,114 @@
+import requests
+from requests.exceptions import HTTPError
+import tempfile
+import subprocess
+from kinto_http import Client, KintoException
+import sys
+
+COMMIT_HASH_URL = "https://api.github.com/repos/publicsuffix/list/commits/master"
+
+LIST_URL = (
+    "https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat"
+)
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"
+
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+CREDENTIALS = ("arpit73", "s3cr3t")  # (username, password)
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Fetching Failed")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash(url):
+    response = requests.get(url)
+    return response.json()["sha"]
+
+
+@handle_request_errors
+def download_resources(*urls, **kwargs):
+    for url in urls:
+        # file_location is found by appending the file_name(at the end of url string) to temp directory
+        file_location = kwargs["directory"] + "/" + url.split("/")[-1]
+        print(file_location)
+        response = requests.get(url, stream=True)
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+                else:
+                    print("Error!!!")
+
+
+# TODO
+def check_last_stored_hash():
+    return True
+
+
+def publish_dafsa(server, credentials, commit_hash):
+
+    bucket_id = "firefox-core-network-dns"
+    collection_id = "public-suffix-list"
+    record_id = "latest-commit-hash"
+
+    client = Client(server_url=server, auth=credentials)
+
+    latest_hash = get_latest_hash(COMMIT_HASH_URL)
+
+    try:
+        record = client.get_record(
+            id=record_id, bucket=bucket_id, collection=collection_id
+        )
+    except KintoException as e:
+        print(e)
+        client.create_record(
+            id=record_id,
+            data={"latest-commit-hash": commit_hash},
+            collection=collection_id,
+            bucket=bucket_id,
+        )
+
+
+    if record["data"]["latest-commit-hash"] == latest_hash:
+        sys.exit()
```

Comment:
Don't use `sys.exit()` here, it will make the function impossible to unit test. `return` should be enough :)